### DISCUSSION TITLE
Question: Comparison with other repos

### DISCUSSION THREAD

**Discussion by BradKML** [DataCanvasIO/HyperTS#57] [General]
How is this repo different from AutoTS? It only seems that this repo has more algorithms on the pipeline? https://winedarksea.github.io/AutoTS/build/html/source/tutorial.html

P.S. This AutoTS tool exists https://github.com/AutoViML/Auto_TS

**Comment by zhangxjohn** [comment]
HyperTS is clearer at the algorithmic level (statistical models,  neural networks, neural architecture search) , as well as at the task level (forecasting, classification, regression, meta-feature extraction)，Automation relies on [Hypernets](https://github.com/DataCanvasIO/Hypernets) (A General Automated Machine Learning Framework to simplify the development of end-to-end AutoML toolkits in specific domains.), modeling is fast, easy to expand, and the API is simple and easy to use.

**Reply by BradKML** [reply]
Model wise, they seem to have less diversity (?)
- https://hyperts.readthedocs.io/en/latest/contents/0700_models.html#statistical-methods
- https://winedarksea.github.io/AutoTS/build/html/source/tutorial.html#models-1
- https://github.com/firmai/atspy#automated-models
- https://opensource.salesforce.com/Merlion/v1.3.0/merlion.models.anomaly.forecast_based.html
- https://hctsa-users.gitbook.io/hctsa-manual/readme/list-of-included-code-files#time-series-model-fitting-and-forecasting

Is it likely that some of the models are generalizations of others?

Also weirdly some have anomaly detection and feature engineering, but that is out of scope for this page.

**Reply by newbei** [reply]
Thanks for your interests. We have investigated the above links. 

In general, our HyperTS pays more attention on the optimization of search space, instead of algorithms diversity. We include the most popular algorithms, and make sure to build well-performed models through advanced and comprehensive search strategies. As a compensation, HyperTS enables user to [build their own algorithms]( https://hyperts.readthedocs.io/en/latest/contents/0600_custom_functions.html).

Furthermore, HyperTS is more concerned on the time efficiency to meet the enterprise demands. Since they always request to achieve better performance on restricted time. 

I hope the answer makes it clear to you. 

You also mentioned about anomaly detection and feature engineering. We are wondering which features of an automated time series tool are more valuable for you.

https://hyperts.readthedocs.io/en/latest/contents/0600_custom_functions.html
